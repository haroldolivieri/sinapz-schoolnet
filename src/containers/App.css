.main-content {
    background-color: "#FAF8F8";
    height: 100vh;
    width: 100vw;
    padding: 0 1em 0 1em;
}
@media only screen and (min-width: 768px) {
    .main-content {
        width: 70vw
    }
}

.app-frame {
    display: flex;
    width: 100%;
}

.toolbar {
    align-items: center;
    display: flex;
    height: 5em;
}