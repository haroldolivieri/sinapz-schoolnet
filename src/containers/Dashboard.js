import React from 'react';
import Grid from "@material-ui/core/Grid";
import { DashboardCard, DashboardPaper } from '../components'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'


export const DashboardIndice = ({
    quantidade = 0,
    tipo = 'Unknown',
}) => (
        <DashboardCard
            quantidade = { quantidade }
            tipo = { tipo }
        />
)
DashboardIndice.propTypes = {
    quantidade: PropTypes.number,
    tipo: PropTypes.string,
}

export const DashboardAvaliacao = ({
    tipo = 'Unknown',
    disciplina = 'Unknown',
    grupo = 'Unknown',
    tempoParaAvaliacao = 'Unknown'
}) => (
    <DashboardPaper
        tipo = { tipo }
        disciplina = { disciplina }
        grupo = { grupo }
        tempoParaAvaliacao = { tempoParaAvaliacao }
    />
)
DashboardAvaliacao.propTypes = {
    tipo: PropTypes.string,
    disciplina: PropTypes.string,
    grupo: PropTypes.string,
    tempoParaAvaliacao: PropTypes.string,
}

const Dashboard = ({
    dashboardIndices = [],
    dashboardAvaliacoes = []
}) => (
    <div>
        <div>
            <Grid container spacing={24}>
            {dashboardIndices.map(item => (
                <Grid item md={4}>
                    <DashboardCard
                        quantidade={item.quantidade}
                        tipo={item.tipo}
                    />
                </Grid>
                ))}
            </Grid>
        </div>
        <div>
            <h1>Próximas Avaliações em 24h </h1>
            {dashboardAvaliacoes.map(item => (
                    <DashboardPaper
                        tipo={item.tipo}
                        disciplina={item.disciplina}
                        grupo={item.grupo}
                        tempoParaAvaliacao={item.tempoParaAvaliacao}
                    />
                ))}
        </div>
    </div>
)

const mapStateToProps = state => ({
    dashboardIndices: state.dashboard.get('dashboardIndices'),
    dashboardAvaliacoes: state.dashboard.get('dashboardAvaliacoes')
})

// function mapStateToProps(state) {
//     dashboardIndices: state.dashboard.get('dashboardIndices')
//     dashboardAvaliacoes: state.dashboard.get('dashboardAvaliacoes')
//     console.log(state)
//   }

export default connect(mapStateToProps)(Dashboard)

Dashboard.propTypes = {
    dashboardIndices: PropTypes.array,
    dashboardAvaliacoes: PropTypes.array
}