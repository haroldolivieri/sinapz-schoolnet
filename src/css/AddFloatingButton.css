.fab {
    position: fixed;
    bottom: 0;
    right: 0;
    margin: 2em;
}

h3 {
    color: white;
}
