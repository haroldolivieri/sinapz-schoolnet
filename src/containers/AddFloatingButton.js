import React, { Component } from 'react'
import ClickAwayListener from '@material-ui/core/ClickAwayListener'
import Fab from '@material-ui/core/Fab'
import Grow from '@material-ui/core/Grow'
import MenuItem from '@material-ui/core/MenuItem'
import MenuList from '@material-ui/core/MenuList'
import Paper from '@material-ui/core/Paper'
import Popper from '@material-ui/core/Popper'
import AddIcon from '@material-ui/icons/Add'
import PropTypes from 'prop-types'
import '../css/AddFloatingButton.css'

export class AddFloatingButton extends Component {
    state = {
        open: false
    }

    toogleFabPopup = () => {
        this.setState(state => ({ open: !state.open }))
    }

    closeFabPopup = () => {
        this.setState({ open: false })
    }

    render() {
        const items = this.props.popupItems

        const popupMenu = (
            <Paper style={{ backgroundColor: '#3c4754' }}>
                <ClickAwayListener onClickAway={this.closeFabPopup}>
                    <MenuList>
                        {items.map(popupItem => (
                            <MenuItem
                                key={popupItem.text}
                                onClick={() => {
                                    popupItem.action()
                                    this.closeFabPopup()
                                }}
                            >
                                <h3>{popupItem.text}</h3>
                            </MenuItem>
                        ))}
                    </MenuList>
                </ClickAwayListener>
            </Paper>
        )

        return (
            <div className="fab">
                <Fab
                    color="primary"
                    buttonRef={node => {
                        this.anchorEl = node
                    }}
                    onClick={
                        items.length === 1
                            ? items[0].action
                            : this.toogleFabPopup
                    }
                >
                    <AddIcon />
                </Fab>
                <Popper
                    open={this.state.open}
                    anchorEl={this.anchorEl}
                    placement="top-end"
                    transition
                >
                    {({ TransitionProps }) => (
                        <Grow {...TransitionProps}>{popupMenu}</Grow>
                    )}
                </Popper>
            </div>
        )
    }
}
AddFloatingButton.propTypes = {
    popupItems: PropTypes.array.isRequired
}
