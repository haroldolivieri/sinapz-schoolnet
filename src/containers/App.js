import React, { Component } from 'react';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ResponsiveDrawer from '../components/ResponsiveDrawer'
import SideDrawer from '../components/SideDrawer';
import '../css/App.css'

import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';

const uitheme = createMuiTheme({
  palette: {
    secondary: { main: '#4986DD' }
  },
  typography: {
    useNextVariants: true,
    fontSize: '11',
    fontFamily: 'Montserrat, sans-serif'
  }
});

class App extends Component {

  state = {
    navDrawerOpened: false
  };

  toggleDrawer = (open) => () => {
    this.setState({ navDrawerOpened: open });
  };

  render() {

    return (
      <MuiThemeProvider theme={uitheme}>

        <div className="app-frame">

          <div className="only-mobile">
            <ResponsiveDrawer
              opened={this.state.navDrawerOpened}
              onClose={this.toggleDrawer}
            />
          </div>

          <div className="hidden-for-mobile">
            <SideDrawer />
          </div>

          <main className="main-content">

            <div className="toolbar">
              <div className="only-mobile">
                <IconButton
                  onClick={this.toggleDrawer(true)}
                  aria-label="Open drawer">
                  <MenuIcon />
                </IconButton>
              </div>
            </div>

            <div>
              {/* Changing depending on route */}
              <Typography paragraph>
                Consequat mauris nunc congue nisi vitae suscipit. Fringilla est ullamcorper eget nulla
                facilisi etiam dignissim diam. Pulvinar elementum integer enim neque volutpat ac
                tincidunt. Ornare suspendisse sed nisi lacus sed viverra tellus. Purus sit amet volutpat
                consequat mauris.
              </Typography>
            </div>

          </main>
        </div>
      </MuiThemeProvider >
    );
  }
}

export default App;
