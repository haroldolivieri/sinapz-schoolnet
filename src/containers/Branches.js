import React from 'react'
import { BranchList } from '../components'
import '../css/Branches.css'
import PropTypes from 'prop-types'

class Branches extends React.Component {
  render() {
    return (
      <div>
        <h1>Filiais</h1>
        <div className=".branch-list-container">
          <BranchList branches={this.props.branches} />
        </div>
      </div>
    )
  }
}

Branches.propTypes = {
  branches: PropTypes.array
}

export default Branches