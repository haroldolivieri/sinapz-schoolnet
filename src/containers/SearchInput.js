import React, { Component } from 'react'
import TextField from '@material-ui/core/TextField'
import SearchRounded from '@material-ui/icons/SearchRounded'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import { debounce } from 'throttle-debounce'
import { fetchSearchResults } from '../searchbar/actions'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Paper from '@material-ui/core/Paper'
import '../css/SearchInput.css'

export const SearchResultItem = ({
    name = 'Unknown',
    description = 'Description',
    photoUrl = 'https://cdn.shopify.com/s/files/1/0539/4361/products/anonymous-guy-fawkes-anarchy-decal.jpg?v=1410346765',
    action = function() {}
}) => (
    <Paper square>
        <ListItem
            className="result-item"
            id="list-item"
            button
            onClick={action}
        >
            <ListItemIcon>
                <img className="result-icon" src={photoUrl} alt={name} />
            </ListItemIcon>
            <div>
                <p className="result-name">{name}</p>
                <p className="result-description">{description}</p>
            </div>
        </ListItem>
    </Paper>
)
SearchResultItem.propTypes = {
    name: PropTypes.string,
    description: PropTypes.string,
    photoUrl: PropTypes.string,
    action: PropTypes.func
}

export class SearchInput extends Component {
    constructor(props) {
        super(props)
        this.autocompleteSearchDebounced = debounce(
            500,
            this.autocompleteSearch
        )
    }

    state = {
        searchInput: ''
    }

    onInputChanged = event => {
        this.setStateForInput(event.target.value)
    }

    onResultClicked = (type, id) => {
        this.clearInput()
        //navigate to right page
    }

    clearInput = () => {
        this.setState({ searchInput: '' }, () => {
            this.autocompleteSearch(this.state.searchInput)
        })
    }

    setStateForInput = searchInput => {
        this.setState({ searchInput }, () => {
            this.autocompleteSearchDebounced(this.state.searchInput)
        })
    }

    autocompleteSearch = input => {
        this.props.fetchSearchResults(input)
    }

    render() {
        return (
            <div>
                <TextField
                    className="input-search"
                    fullWidth={true}
                    value={this.state.searchInput}
                    onChange={this.onInputChanged}
                    onBlur={this.clearInput}
                    label="Busque por alunos, professores e filiais"
                    InputProps={{
                        endAdornment: <SearchRounded />
                    }}
                />

                {this.props.results.length > 0 && (
                    <div className="results">
                        {this.props.results.map(item => (
                            <SearchResultItem
                                key={item.name}
                                name={item.name}
                                description={item.description}
                                action={this.onResultClicked.bind(
                                    this,
                                    item.type,
                                    item.id
                                )}
                            />
                        ))}
                    </div>
                )}
            </div>
        )
    }
}

const mapStateToProps = state => ({
    results: state.searchBar.get('searchResults')
})

const mapDispatchToProps = dispatch =>
    bindActionCreators({ fetchSearchResults }, dispatch)

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(SearchInput)

SearchInput.propTypes = {
    results: PropTypes.array.isRequired,
    fetchSearchResults: PropTypes.func.isRequired
}
