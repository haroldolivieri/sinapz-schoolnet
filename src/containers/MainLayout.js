import React, { Component } from 'react'
import MenuIcon from '@material-ui/icons/Menu'
import { ResponsiveDrawer, SideDrawer } from '../components'
import { loadInitalData } from '../menu/actions'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import SearchInput from '../containers/SearchInput'
import '../css/MainLayout.css'

export class MainLayout extends Component {
    constructor(props) {
        super(props)

        this.props.loadInitalData()
    }

    state = {
        navDrawerOpened: false,
        selectedItem: 'dashboard'
    }

    toggleDrawer = () => {
        this.setState(state => ({ navDrawerOpened: !state.navDrawerOpened }))
    }

    renderToolbar() {
        return (
            <div className="toolbar">
                <MenuIcon
                    className="only-mobile"
                    onClick={this.toggleDrawer}
                    aria-label="Open drawer"
                />

                <img
                    className="logo"
                    alt="logo"
                    src={require(`../images/logo.png`)}
                />

                <div className="search">
                    <SearchInput />
                </div>
            </div>
        )
    }

    render() {
        return (
            <div className="app-frame">
                <div className="only-mobile">
                    <ResponsiveDrawer
                        opened={this.state.navDrawerOpened}
                        onClose={this.toggleDrawer}
                    />
                </div>

                <div className="hidden-for-mobile">
                    <SideDrawer />
                </div>

                {/* Toolbar with hamburguer menu, logo and search input */}
                {this.renderToolbar()}

                {/* Content injected by router */}
                <div className="children-content">{this.props.children}</div>
            </div>
        )
    }
}

const mapDispatchToProps = dispatch =>
    bindActionCreators({ loadInitalData }, dispatch)
export default connect(
    null,
    mapDispatchToProps
)(MainLayout)
