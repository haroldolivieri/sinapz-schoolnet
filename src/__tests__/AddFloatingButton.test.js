import React from 'react'
import { shallow } from 'enzyme'
import { AddFloatingButton } from '../containers/AddFloatingButton'
import { Fab } from '@material-ui/core'

describe('AddFloatingButton', () => {
    let actionFirstItem
    let actionSecondItem
    let actionThirdItem
    let items
    beforeEach(() => (actionFirstItem = jest.fn()))
    beforeEach(() => (actionSecondItem = jest.fn()))
    beforeEach(() => (actionThirdItem = jest.fn()))
    beforeEach(
        () =>
            (items = [
                {
                    text: 'Adicionar Evento',
                    action: actionFirstItem
                },
                {
                    text: 'Adicionar Filial',
                    action: actionSecondItem
                },
                {
                    text: 'Adicionar Professor',
                    action: actionThirdItem
                }
            ])
    )

    describe('renders', () => {
        let wrapper
        beforeEach(
            () => (wrapper = shallow(<AddFloatingButton popupItems={items} />))
        )

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot()
        })

        it('renders correctly after click', () => {
            wrapper.find(Fab).simulate('click')
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe('has only one item', () => {
        let wrapper
        beforeEach(
            () =>
                (wrapper = shallow(
                    <AddFloatingButton popupItems={[items[0]]} />
                ))
        )

        it('click on fab should call first item action', () => {
            wrapper.find(Fab).simulate('click')

            expect(actionFirstItem).toHaveBeenCalledTimes(1)
        })

        it('click on fab item should not open popup', () => {
            wrapper.find(Fab).simulate('click')

            expect(wrapper.state().open).toBeFalsy()
        })
    })

    describe('has more than one item', () => {
        let wrapper
        beforeEach(
            () => (wrapper = shallow(<AddFloatingButton popupItems={items} />))
        )

        it('click on fab should not call first item action', () => {
            wrapper.find(Fab).simulate('click')

            expect(actionFirstItem).toHaveBeenCalledTimes(0)
        })

        it('click on fab should open popup', () => {
            wrapper.find(Fab).simulate('click')

            expect(wrapper.state().open).toBeTruthy()
        })
    })
})
