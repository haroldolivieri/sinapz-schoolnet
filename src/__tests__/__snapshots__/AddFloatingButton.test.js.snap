// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFloatingButton renders renders correctly many items 1`] = `
<div
  className="fab"
>
  <WithStyles(Fab)
    buttonRef={[Function]}
    color="primary"
    onClick={[Function]}
  >
    <pure(AddIcon) />
  </WithStyles(Fab)>
  <Popper
    disablePortal={false}
    open={true}
    placement="top-end"
    transition={true}
  >
    <Component />
  </Popper>
</div>
`;

exports[`AddFloatingButton renders renders correctly no items 1`] = `
<div
  className="fab"
>
  <WithStyles(Fab)
    buttonRef={[Function]}
    color="primary"
    onClick={[Function]}
  >
    <pure(AddIcon) />
  </WithStyles(Fab)>
  <Popper
    disablePortal={false}
    open={false}
    placement="top-end"
    transition={true}
  >
    <Component />
  </Popper>
</div>
`;
